docs/content/sips/008-logging.md

```diff
@@ -0,0 +1,104 @@
+title = "SIP 008 - Runtime Logging"
+template = "main"
+date = "2022-12-09T11:36:36Z"
+---
+
+Summary: Unified runtime logging
+
+Owner: mail@etehtsea.me
+
+Created: December 09, 2022
+
+## Background
+
+## Current experience
+
+Currently, by default spin prints into `stdout` only the basic info:
+
+```
+$ spin up
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  hello: http://127.0.0.1:3000/hello
+    A simple component that returns hello.
+```
+
+Application log is implicitly written into `~/.spin/<spin_app>/logs/<component_id>_std{err|out}.log`.
+The log for the spin itself has log level `ERROR` and outputs into `stderr`.
+
+To enable a full logging experience, a developer needs to provide `--follow-all` to duplicate app log into `stderr` and also `RUST_LOG=spin=<log_level>` to see the spin log in the `stderr`.
+
+
+## Proposal
+
+- Set default log level to `INFO`.
+- Set default log destination to `stdout/stderr`.
+- Provide a way for spin apps for logging (besides using `println!`/`eprintln!`).
+- Log `WARN`and `ERROR` into `stderr` and `INFO`, `DEBUG`, `TRACE` into `stdout`.
+- Log level `OFF` disables logging completely.
+
+### Goal
+
+To provide the new baseline that gives foundation for further iterative development of new features and enhancements based on the feedback.
+
+### Configuration
+
+Command-line arguments:
+
+```
+--log-level <LOG_LEVEL>
+    Describes the level of verbosity
+
+    [default: INFO]
+    [possible values: ERROR, WARN, INFO, DEBUG, TRACE, OFF]
+
+--log-timezone <LOG_TIMEZONE>
+    [possible values: local, utc]
+```
+
+`spin.toml`:
+```toml
+[variables]
+log = {
```

Comment:
Not quite sure what you're envisaging here.  Are you thinking of the runtime config file rather than the application definition?  I'm not quite seeing how this fits with the current use of the app def `variables` section.

Reply:
The initial thought was that when log configuration evolves, it gets pretty inconvenient to provide dozens command line arguments, so it has to be convenient to define it in the config.

Also, I've found a reference to it in the [app-config SIP](https://github.com/fermyon/spin/blob/main/docs/content/sips/002-app-config.md#components-and-applications-set-configuration-values-of-their-dependencies).

But I didn't take into consideration that `spin.toml` is somewhat static and is bundled into bindle artifact. And there is no dev/prod configuration at the moment.

I think I better move this into `Future design considerations`.

Reply:
It might fit in the new `runtime-config.toml` file but I don't know much about the intended scope of that... future considerations sounds good to me _grin_